experimental/shared/src/main/scala/zio/prelude/experimental/ApplicationCompose.scala

```diff
@@ -0,0 +1,53 @@
+package zio.prelude
+package experimental
+
+trait ApplicationCompose[:=>[-_, +_]] extends BothCompose[:=>] {
+
+  type :-->[-_, +_]
```

Comment:
This is a lot of generality.  What are motivating cases to support `:-->` and `:=>` simultaneously?

Reply:
I don't know, to be honest. I just know that that's the way CCC is usually defined (in Haskell).
https://hackage.haskell.org/package/categories-1.0.7/docs/Control-Category-Cartesian-Closed.html
I learned about this topic here, btw https://youtu.be/SVRYcrhRCes?t=449